dali/python/nvidia/dali/pipeline.py

```diff
@@ -77,14 +75,12 @@ class Pipeline(object):
         Negative values for this parameter are invalid - the default
         value may only be used with serialized pipeline (the value
         stored in serialized pipeline is used instead).
-    device_id : int, optional, default = -1
+    device_id : int, optional, default = None
         Id of GPU used by the pipeline.
-        A None value for this parameter means that DALI should not use GPU nor CUDA runtime.
-        This limits the pipeline to only CPU operators but allows it to run on any CPU capable
-        machine.
-    seed : int, optional, default = -1
-        Seed used for random number generation. Leaving the default value
-        for this parameter results in random seed.
+        If the pipeline requires a GPU and this field is left unspecified, DALI will use the
+        current CUDA device (according to ``cudaGetDevice``) at the time the pipeline is built.
+    seed : int, optional, default = None
+        Seed used for random number generation. If not set, a random seed will be generated.
```

Comment:
Did you happen to check, how this change will impact existing pipelines that did not specify `seed` argument?

Reply:
Negative seeds are translated to `nullopt` upon deserialization - see `pipeline.cc:180`.

Reply:
...as of today, if you don't specify a seed and then serialize then the time-based seed is selected before serialization. This is almost certainly a bug as I can see that we set non-negative seeds when we test seed serialization (as if leaving it negative would defeat it). With that, it's impossible now to deserialize a pipeline and have a random seed each time it's created.